### PR TITLE
chore: generate separate versions jsons for modules

### PR DESCRIPTION
**Pull request by ZheSun88**
waiting for https://github.com/vaadin/flow/pull/14014

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,6 +7,8 @@ error-screenshots/
 .project
 .settings
 tmp.*
+vaadin-core/vaadin_versions_core.json
```
Comment (taefi) [inline .gitignore:10]:
```suggestion
vaadin-core/vaadin-core-versions.json
```

File: .gitignore [hunk 2]
```diff
@@ -7,6 +7,8 @@ error-screenshots/
...
+vaadin/vaadin_versions.json
```
Comment (taefi) [inline .gitignore:11]:
```suggestion
vaadin/vaadin-versions.json
```

File: scripts/generateBoms.sh [hunk 3]
```diff
@@ -26,3 +26,6 @@ cp scripts/generator/results/hilla-maven-plugin-pom.xml hilla-maven-plugin/pom.x
 cp scripts/generator/results/vaadin-gradle-plugin-pom.xml vaadin-gradle-plugin/pom.xml
 cp scripts/generator/results/vaadin-gradle-plugin-portal-pom.xml vaadin-gradle-plugin/pom-portal.xml
 cp scripts/generator/results/vaadin-platform-servlet-containers-tests-pom.xml vaadin-platform-servlet-containers-tests/pom.xml
+cp scripts/generator/results/vaadin-versions-core.json vaadin-core/vaadin-versions-core.json
```
Comment (taefi) [inline scripts/generateBoms.sh:29]:
```suggestion
cp scripts/generator/results/vaadin-core-versions.json vaadin-core/vaadin-core-versions.json
```

File: scripts/generator/generate.js [hunk 4]
```diff
@@ -28,6 +28,12 @@ function getResultsFilePath(filename) {
     return path.resolve(`${__dirname}/results/${filename}`);
 }
 
+const coreJsonTemplateFileName = getTemplateFilePath('template-vaadin-core-versions.json');
+const vaadinCoreJsonFileName = getResultsFilePath('vaadin-versions-core.json');
```
Comment (taefi) [inline scripts/generator/generate.js:32]:
```suggestion
const vaadinCoreJsonFileName = getResultsFilePath('vaadin-core-versions.json');
```

File: scripts/generator/templates/template-vaadin-core-versions.json [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
+	
+}
```
Comment (taefi) [inline scripts/generator/templates/template-vaadin-core-versions.json:3]:
```suggestion
}

```

File: scripts/generator/templates/template-vaadin-versions.json [hunk 6]
```diff
@@ -0,0 +1,3 @@
+{
+	
+}
```
Comment (taefi) [inline scripts/generator/templates/template-vaadin-versions.json:3]:
```suggestion
}

```

File: vaadin-core/src/test/java/VersionsFileTest.java [hunk 7]
```diff
@@ -27,7 +27,7 @@ public class VersionsFileTest {
     public void testVersionsFile_existsAsClassPathResource()
             throws IOException {
         File file = new File(
-                getClass().getResource("vaadin_versions.json").getFile());
+                getClass().getResource("vaadin_versions_core.json").getFile());
```
Comment (taefi) [inline vaadin-core/src/test/java/VersionsFileTest.java:30]:
```suggestion
                getClass().getResource("vaadin-core-versions.json").getFile());
```

File: vaadin-core/src/test/java/VersionsFileTest.java [hunk 8]
```diff
@@ -27,7 +27,7 @@ public class VersionsFileTest {
...
         Assert.assertTrue("vaadin_versions.json file is missing",
```
Comment (taefi) [inline vaadin-core/src/test/java/VersionsFileTest.java:31]:
```suggestion
        Assert.assertTrue("vaadin-core-versions.json file is missing",
```

**Review by taefi — CHANGES_REQUESTED** [review]
I suggest to change the `-versions` files naming to comply with the artifact name:
```
vaadin      -> vaadin-versions.json
vaadin-core -> vaadin-core-versions.json
```

**Review by taefi — APPROVED** [review]

**Review by manolo — APPROVED** [review]

**Pull request merged**